api/resources_portal/test/views/test_grant.py

```diff
@@ -0,0 +1,122 @@
+from django.forms.models import model_to_dict
+from django.urls import reverse
+from rest_framework import status
+from rest_framework.test import APITestCase
+
+from faker import Faker
+
+from resources_portal.models import Material, User
+from resources_portal.test.factories import (
+    GrantFactory,
+    LeafGrantFactory,
+    MaterialFactory,
+    UserFactory,
+)
+
+fake = Faker()
+
+
+class TestGrantPostTestCase(APITestCase):
+    """
+    Tests /grants list operations.
+    """
+
+    def setUp(self):
+        self.url = reverse("grant-list")
+        self.grant = LeafGrantFactory()
+
+    def test_post_request_with_no_data_fails(self):
+        response = self.client.post(self.url, {})
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_post_request_with_valid_data_succeeds(self):
+        grant_data = model_to_dict(self.grant)
+
+        response = self.client.post(self.url, grant_data, format="json")
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+
+class TestSingleGrantTestCase(APITestCase):
+    """
+    Tests /grants detail operations.
+    """
+
+    def setUp(self):
+        self.grant = GrantFactory()
+        self.url = reverse("grant-detail", args=[self.grant.id])
+
+    def test_get_request_returns_a_given_grant(self):
+        response = self.client.get(self.url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_put_request_updates_a_grant(self):
+        grant_json = self.client.get(self.url).json()
+
+        new_title = "New Title"
+        new_funder_id = "YYY-YYY-YYY"
+        grant_json["title"] = new_title
+        grant_json["funder_id"] = new_funder_id
+
+        # Test that users won't be updated.
+        new_member = UserFactory()
+        new_member_json = {"id": new_member.id}
+        grant_json["users"].append(new_member_json)
+
+        # TODO: this should require authentication
```

Comment:
You can add this in your code as-is. To authenticate during your tests, add

`self.client.force_authenticate(user=self.grant.users.first())`
in your test set-up. To check authentication on the ViewSet, override the create() method of your grant view and add the lines

`if not request.user.is_authenticated:`
`     return Response(status=status.HTTP_403_FORBIDDEN)`.

Or at least this is how I did it in [the invitations route](https://github.com/AlexsLemonade/resources-portal/pull/152). If there's a better way, let me know. If you end up doing this you'll also want to add a test for unauthenticated users being rejected.